core/unit_test/TestComplex.hpp

```diff
@@ -30,6 +29,9 @@
 #endif
 #endif
 
+#include <Kokkos_Core.hpp>
```

Comment:
Why did we move these headers below the pragma push?

Reply:
See pull request description. We need to have the push before the problematic headers.

Reply:
Then at least do comment that this is not enough to pragma push around the call site in this file that triggers all the instantiations but that we need these suppression actually applied on our headers.

:(